### PR TITLE
sandbox testing fixes

### PR DESCRIPTION
**Pull request by muhammad-ammar**
Fixes required for veda sandbox built using Ansible.

@yro @Qubad786 Please review these changes.

### CODE REVIEW

File: control/veda_file_discovery.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ def discover_studio_ingested_videos(self):
                 connection = boto.connect_s3()
                 self.bucket = connection.get_bucket(self.auth_dict['edx_s3_ingest_bucket'])
                 for video_s3_key in self.bucket.list(self.auth_dict['edx_s3_ingest_prefix'], '/'):
-                    if video_s3_key.name != 'prod-edx/unprocessed/':
+                    if video_s3_key.name != self.auth_dict['edx_s3_ingest_prefix']:
                         self.validate_metadata_and_feed_to_ingest(video_s3_key=self.bucket.get_key(video_s3_key.name))
```
Comment (yro) [inline control/veda_file_discovery.py]: Nice.

File: VEDA_OS01/fixtures/encodes.json [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        }
+    },
+    {
+        "model": "VEDA_OS01.encode",
+        "pk": 3,
+        "fields": {
+            "encode_destination": 1,
+            "encode_name": "Low Bandwidth Override",
+            "profile_active": true,
+            "encode_suffix": "LBO",
```
Comment (yro) [inline VEDA_OS01/fixtures/encodes.json]: This can be false.

File: VEDA_OS01/fixtures/encodes.json [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        }
+    },
+    {
+        "model": "VEDA_OS01.encode",
+        "pk": 8,
+        "fields": {
+            "encode_destination": 4,
+            "encode_name": "Mobile - High",
+            "profile_active": false,
+            "encode_suffix": "MB1",
```
Comment (yro) [inline VEDA_OS01/fixtures/encodes.json]: this should be true (it's set false by mistake)

**Comment by yro** [comment]
@muhammad-ammar Sorry - I commented on the fixtures before I saw "sandbox". Ignore those (deleted)
This looks good!
👍

File: control/control_env.py [hunk 4]
```diff
@@ -25,14 +25,18 @@
...
 Central Config
 """
-WORK_DIRECTORY = os.path.join(
+CONFIG = get_config()
+
+DEFAULT_WORK_DIRECTORY = os.path.join(
     os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
     'VEDA_WORKING'
 )
+WORK_DIRECTORY = CONFIG.get('VEDA_WORKING', DEFAULT_WORK_DIRECTORY)
```
Comment (Qubad786) [inline control/control_env.py]: can we change `VEDA_WORKING` config?

File: instance_config.yaml [hunk 5]
```diff
@@ -121,3 +121,6 @@ sg_script_key:
 # Endpoints
 # ---
 threeplay_ftphost:
+
+lms_base_url: 'http://127.0.0.1'
```
Comment (Qubad786) [inline instance_config.yaml]: lets leave this empty.

File: instance_config.yaml [hunk 6]
```diff
@@ -98,8 +98,8 @@ val_api_url:
 val_token_url:
 val_video_images_url:
 # Credentials
-val_client_id:
-val_secret_key:
+val_client_id: clientkey
```
Comment (Qubad786) [inline instance_config.yaml]: Lets add quotes for consistency.

File: VEDA/tests/test_utils.py [hunk 7]
```diff
@@ -56,6 +58,22 @@ def setUp(self):
         with open(self.static_file_path, 'w') as outfile:
             yaml.dump(TEST_STATIC_CONFIG, outfile, default_flow_style=False)
 
+        # create a temp directory with temp directories and files in it
```
Comment (Qubad786) [inline VEDA/tests/test_utils.py]: Lets keep further utils tests separate from this test case.

File: VEDA/settings/production.py [hunk 8]
```diff
@@ -28,3 +28,10 @@
         vars()[key].update(value)
 
 vars().update(CONFIG_DATA)
+
+JWT_AUTH = {
+    'JWT_SECRET_KEY': CONFIG_DATA['val_secret_key'],
+    'JWT_ISSUER': '{}/oauth2'.format(CONFIG_DATA['lms_base_url']),
```
Comment (Qubad786) [inline VEDA/settings/production.py]: we may need to `rstrip` the `CONFIG_DATA['lms_base_url']` too.

**Comment by muhammad-ammar** [comment]
@Qubad786 addressed your feedback

File: control/control_env.py (hunk 4 above)
Comment (muhammad-ammar) [reply control/control_env.py]: for now I will prefer to leave it as is.

File: VEDA_OS01/models.py [hunk 9]
```diff
@@ -540,7 +540,7 @@ class Destination(models.Model):
     destination_nick = models.CharField('Nickname (3 Char.)', max_length=3, null=True, blank=True)
 
     def __unicode__(self):
-        return u'%s'.format(self.destination_name) or u''
+        return u'{}'.format(self.destination_name)
```
Comment (Qubad786) [inline VEDA_OS01/models.py]: dont need to format, we can just return it.

File: VEDA_OS01/tests/test_transcripts.py [hunk 10]
```diff
@@ -1565,3 +1558,27 @@ def test_translation_retrieval_with_invalid_course_id(self, course_runs, mock_lo
...
+        Tests setup.
+        """
+        super(TranscriptNameConstructionTests, self).setUp()
+
+    def test_upload_sjson_to_s3(self):
+        """
+        Verify that `construct_transcript_names` works as expected.
+        """
+        edxval_name, s3_name = transcripts.construct_transcript_names(CONFIG_DATA)
+        self.assertNotEqual(
```
Comment (Qubad786) [inline VEDA_OS01/tests/test_transcripts.py]: this is not necessarily true (i.e. for prod).

Comment (Qubad786) [reply VEDA_OS01/tests/test_transcripts.py]: Rather we can check that it contains/or not contains instance prefix.

**Pull request merged**